internal/services/pam/pam.go

```diff
@@ -240,6 +240,15 @@ func (s Service) IsAuthenticated(ctx context.Context, req *authd.IARequest) (res
 			return nil, fmt.Errorf("user data from broker invalid: %v", err)
 		}
 
+		selectedUsername, err := s.brokerManager.UsernameFromSessionID(sessionID)
```

Comment:
Initially, yes, but it felt bad to have another layer of `map[id]session`. I can refactor it in a new commit so we can see how it looks.

Reply:
Actually, I'll just reset it since it also does feel bad to have the broker manager dealing with session management

Reply:
So, it means that you want to have the broker object itself dealing with the username cache per session in internal/brokers/?

This will be cleaner indeed, but it means that you have 2 maps (one on the manager for session id to broker) and one on the broker for session id to username. That will complexify it a little bit too much for you, no?

But I agree this is logically cleaner (and we don’t leak the username retrieval per session in internal/services/pam at all this way)

Reply:
> That will complexify it a little bit too much for you, no?

Not exactly sure what you meant, but I don't think so. It's just a question of remembering the username when creating a session (and since everything will be done internally, we don't need to expose anything)